cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -2994,920 +3005,146 @@ endmacro()
 # Dependencies for Arrow Flight RPC
 
 macro(build_absl)
-  message(STATUS "Building Abseil-cpp from source")
-  set(absl_FOUND TRUE)
-  set(absl_VERSION ${ARROW_ABSL_BUILD_VERSION})
+  message(STATUS "Building Abseil from source using FetchContent")
```

Comment:
Could you also add `list(APPEND CMAKE_MESSAGE_INDENT "Abseil: ")` and `list(POP_BACK CMAKE_MESSAGE_INDENT)` for better log messages like we did in `build_boost()`:

https://github.com/apache/arrow/blob/d1e6d6bd4b6d32d71bd02123342561b4db59da3e/cpp/cmake_modules/ThirdpartyToolchain.cmake#L1058
https://github.com/apache/arrow/blob/d1e6d6bd4b6d32d71bd02123342561b4db59da3e/cpp/cmake_modules/ThirdpartyToolchain.cmake#L1186